tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,19 +71,1106 @@ public function test_wp_enqueue_script() {
 		$this->assertSame( '', get_echo( 'wp_print_scripts' ) );
 	}
 
+	/**
+	 * Gets delayed strategies as a data provider.
+	 *
+	 * @return array[] Delayed strategies.
+	 */
+	public function data_provider_delayed_strategies() {
+		return array(
+			'defer' => array( 'defer' ),
+			'async' => array( 'async' ),
+		);
+	}
+
+	/**
+	 * Test inline scripts in the `after` position with delayed main script.
+	 *
+	 * If the main script with delayed loading strategy has an `after` inline script,
+	 * the inline script should not be affected.
+	 *
+	 * @ticket 12009
+	 *
+	 * @covers WP_Scripts::do_item
+	 * @covers WP_Scripts::get_inline_script_tag
+	 * @covers ::wp_print_delayed_inline_script_loader
+	 * @covers ::wp_add_inline_script
+	 * @covers ::wp_enqueue_script
+	 *
+	 * @dataProvider data_provider_delayed_strategies
+	 * @param string $strategy Strategy.
```

Comment:
```suggestion
	 * @dataProvider data_provider_delayed_strategies
	 *
	 * @param string $strategy Strategy.
```

Nit-pick, Per [56070](https://core.trac.wordpress.org/ticket/56070) add a line break between two annotation.